MetasysServices/Alarms/AlarmServiceProvider.cs

```diff
@@ -114,9 +159,90 @@ public async Task<IEnumerable<AlarmAnnotation>> GetAnnotationsAsync(Guid alarmId
                 catch (Exception e)
                 {
                     throw new MetasysObjectException(token.ToString(), e);
-                }              
+                }
             }
             return annotationsList;
         }
+
+
+        private async Task<Alarm> CreateItem(JToken item)
+        {
+            Alarm alarm = new Alarm();
+            string unitsUrl = string.Empty;
+            string typeUrl = string.Empty;
+            string categoryUrl = string.Empty;
+
+            try
+            {
+                alarm.Id = (Guid)item["id"];
+                alarm.Self = item["self"].Value<string>();
+                alarm.ItemReference = item["itemReference"].Value<string>();
+                alarm.Name = item["name"].Value<string>();
+                alarm.Message = item["message"].Value<string>();
+                alarm.IsAckRequired = item["isAckRequired"].Value<bool>();
+                alarm.Priority = item["priority"].Value<int>();
+                alarm.CreationTime = item["creationTime"].Value<string>();
+                alarm.IsAcknowledged = item["isAcknowledged"].Value<bool>();
+                alarm.IsDiscarded = item["isDiscarded"].Value<bool>();
+                alarm.ObjectUrl = item["objectUrl"].Value<string>();
+                alarm.AnnotationsUrl = item["annotationsUrl"].Value<string>();
+
+                if (Version < ApiVersion.v3)
+                {
+                    typeUrl = item["typeUrl"].Value<string>();
+                    categoryUrl = item["categoryUrl"].Value<string>();
+                    var unitValue = item["triggerValue"]["unitsUrl"] ?? item["triggerValue"]["valueEnumMemberUrl"];
+                    unitsUrl = unitValue.Value<string>();
+                }
+                else
+                {
+                    alarm.Type = item["type"].Value<string>();
+                    alarm.Category = item["category"].Value<string>();
+                    var unitVar = item["triggerValue"].Contains("units") ? item["triggerValue"]["units"].Value<string>() : null;

```

Comment:
I tested the method to Get Alarms using v3 and I got an error ('ArgumentNullException'). I checked an it seems coming from the function 'CreateItem' due to the variable unitVar is null.
I think it could be fixed writing something like:
```suggestion
                    var unitVar = item["triggerValue"]["units"] ?? item["triggerValue"]["value"];
                    unitKey = unitVar.Value<string>() ?? string.Empty;
                    var measurement = new Measurement
                    {
                        Units = ResourceManager.Localize(unitKey, _CultureInfo),
                        Value = item["triggerValue"]["value"].Value<string>()
                    };
```